src/monaca.js

```diff
@@ -1451,51 +1452,69 @@
 
     fs.exists(projectDir, function(exists) {
       if (exists) {
-        var files = {},
+        let files = {},
           promises = [],
           filteredList = [];
 
-        glob.sync("**/*",
-          {
-            cwd: path.resolve(projectDir),
-            dot: true
+        let framework = '';
+
+        try {
+          // create .monacaignore if there isn't
+          if (!options || !options.framework) {
+            framework = 'cordova';
+          } else {
+            framework = options.framework;
           }
-        ).forEach(
-          function(item) {
-            if (fs.existsSync(path.resolve(projectDir, item))) {
-              filteredList.push(item);
+          let monacaignore = path.resolve(projectDir, '.monacaignore');
+          if (!fs.existsSync(monacaignore)) this._getMonacaIgnore(projectDir, framework);
+          let ignoreList = this._filterMonacaIgnore(projectDir, framework);
```

Comment:
I noticed `framework` is passed to the function. As I read through the code, it is added in case we need to generate `.monacaignore` and that it depend on the framework. However, the current `.monacaignore` is the same between two frameworks.

```
diff -y ./src/default-config/cordova/.monacaignore ./src/default-config/react-native/.monacaignore
node_modules							node_modules
.git								.git
.monaca/*							.monaca/*
!.monaca/project_info.json					!.monaca/project_info.json
android/build							android/build
ios/build							ios/build
build								build
.DS_Store							.DS_Store
```

Therefore, I think we should also remove `framework` parameter to `filterMonacaIgnore` function, and have just one template for `.monacaignore`. Could you please do that in another PR?

Reply:
@masahirotanaka @yong-asial 
The ignore file for Cordova and React Native are not exactly the same. It appears that the Cordova default file was not configured correctly.

React Native projects do not have a `platform` directory unlike Cordova, so the Android and iOS folder are located at the root directory. This is why `android/build` and `ios/build` exists.

For React native, it should have:
```
android/build
android/app/build
ios/build
```
I believe `android/app/build` should also be added to the ignore list. This is the actual output directory for the generated contents, such as apk etc.

Cordova has the `platforms` directory, so using the above ignore list would always fail and upload the build contents.

Cordova should have:
```
platforms/android/build
platforms/android/app/build
platforms/ios/build
```

The iOS will need validation.